Full coverage for CODEOWNERS in the codebase.

DubeySandeep opened the pull request:
This PR adds code owners to all possible files in the codebase.
**Current status:**
- [x] Full coverage, all the files which are tracked by git.
- [x] Update the lint test for the new behavior.

DubeySandeep committed:
Added codeowners for controllers and other backend files.

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -48,11 +52,15 @@
 
 
 # Dev workflow team.
+/core/templates/dev/head/css/.stylelintrc @apb7
```
I have created an issue for this at #6615

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -142,29 +150,36 @@
 /core/templates/dev/head/pages/questions_list/ @aks681
 
+# Practice session project.
+/core/templates/dev/head/pages/practice_session/ @aks681
```
Assuming this to be a part of the new structure project.

seanlip replied:
Perhaps @vinitamurthi should be co-codeowner? Comes under questions.

vinitamurthi replied:
Sure, FYI - I am also adding /core/templates/dev/head/pages/question_player/ in #6620

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -142,29 +150,36 @@
+/core/controllers/practice_sessions*.py @aks681
+
 # Skill project.
+/core/controllers/concept_card_viewer*.py @aks681
```
@aks681 I'm not sure where to place this (As this is also used in the exploration) so can you confirm that we can place this over here?

aks681 replied:
Yeah, it's fine to place it here.

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -173,13 +188,18 @@
 
 
 # Infrastructure.
+/core/controllers/cron*.py @seanlip
```
@seanlip, should I create a different section for `corn, jobs and task` or we can place this over here (inside Infrastructure)? &

seanlip replied:
I think infrastructure is fine.

DubeySandeep committed:
Added codeowners for custom_landing_pages controller.

DubeySandeep commented:
@seanlip & @YashJipkate, is there any specific reason for excluding all the files in the root directory from the codeowner tests? (I'm following CODEOWNERS  warning message for assigning codeowners to the remaining files.)

seanlip commented on .github/CODEOWNERS:
```diff
@@ -48,11 +52,14 @@
 # Exploration project.
 /core/controllers/editor*.py @DubeySandeep
+/core/controllers/reader*.py @aks681
+/core/controllers/resources*.py @seanlip
```
@vojtechjelinek can take this one, I think.

vojtechjelinek replied:
Yeah, sure.

DubeySandeep replied:
Done!

seanlip commented:
> @seanlip & @YashJipkate, is there any specific reason for excluding all the files in the root directory from the codeowner tests? (I'm following CODEOWNERS warning message for assigning codeowners to the remaining files.)

This sounds like an error; we should extend the tests to those files too.

apb7 commented:
Hi @DubeySandeep, please bring your PR branch up-to-date with develop. Thanks!

DubeySandeep committed:
Merge branch 'develop' of https://github.com/oppia/oppia into full-codeowners

DubeySandeep committed:
Added full coverage for the codeowners.

DubeySandeep committed:
Merge branch 'develop' of https://github.com/oppia/oppia into full-codeowners

DubeySandeep committed:
Merge branch 'full-codeowners' of https://github.com/DubeySandeep/oppia into full-codeowners

DubeySandeep committed:
Added full coverage for the codeowners.

DubeySandeep committed:
Update dbranch after makingf delete to rename structure.

DubeySandeep committed:
Codeowners for root directories and associated test cahnges.

DubeySandeep committed:
Updated the tests and made changes as asked by the reviewers.

DubeySandeep committed:
Merge branch 'develop' of https://github.com/oppia/oppia into full-codeowners

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -48,11 +64,27 @@
+/.gitignore @apb7
+/.isort.cfg @apb7
+/.pylintrc @apb7
+/.stylelintrc @apb7
```
@apb7, why do we have two .stylelintrc files? is there any specific reason?

DubeySandeep replied:
@apb7, can you please check this one?

apb7 replied:
I think we had a discussion on this some time back, @DubeySandeep. We have two stylelintrc files -- one for oppia.css and other for css within html files. The two config files use different pre-processors and therefore, are not the same and one cannot be used for the other. Thanks!

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -244,15 +386,22 @@
 /core/templates/dev/head/pages/header_js_libs.html @vojtechjelinek
+/gulpfile.js @vojtechjelinek
+/jinja_utils*.py @vojtechjelinek
+/Makefile @vojtechjelinek
```
@vojtechjelinek, is there any need of this file?  (Introduced at #5659)

vojtechjelinek replied:
No, I was just playing with it and accidentaly commited the file.

DubeySandeep replied:
I've removed this file!

DubeySandeep committed:
Added self-review changes, like deleting unwwanted variables, adding proper spaces and newlines.

DubeySandeep committed:
Added proper docstring.

DubeySandeep commented:
@YashJipkate, Can you please help to review the test pre_commit_linter test file?

DubeySandeep commented:
@seanlip, I have addressed the review changes PTAL! :)

Note: Currently, it takes around 8-10 sec to pass the full codebase code-owners test but we can make it more simple for pre_push hook and I've created an issue for the same at #6654.

seanlip commented on scripts/pre_commit_linter.py:
```diff
@@ -471,6 +466,54 @@ def _get_expression_from_node_if_one_exists(
+
+    Args:
+        root: str. The path from where the function should start walking.
+        exclude_dirs: list(str). A list of dir path which should to be ignored.
```
Drop "to"

DubeySandeep replied:
Done!

seanlip commented on scripts/pre_commit_linter.py:
```diff
@@ -471,6 +466,54 @@ def _get_expression_from_node_if_one_exists(
+        exclude_dirs: list(str). A list of dir path which should to be ignored.
+
+    Yields:
+        str. list(str). A list of unignored files.
```
str or list(str)? Pick one.

DubeySandeep replied:
list(str). Fixed.

seanlip commented on scripts/pre_commit_linter.py:
```diff
@@ -471,6 +466,54 @@ def _get_expression_from_node_if_one_exists(
+    Yields:
+        str. list(str). A list of unignored files.
+    """
+    dirs, nondirs = [], []
```
Does nondirs mean filepaths? If so perhaps that would be a better name.

DubeySandeep replied:
Done!

seanlip commented on scripts/pre_commit_linter.py:
```diff
@@ -471,6 +466,54 @@ def _get_expression_from_node_if_one_exists(
+    """
+    command = ['git', 'check-ignore', '-q', path_to_check]
+
+    # The "git check-ignore <path>" command use to return 0 when the path is
```
use to return --> returns

DubeySandeep replied:
Done!

seanlip commented on scripts/pre_commit_linter.py:
```diff
@@ -2075,93 +2135,36 @@ def _check_codeowner_file(self):
-                            print ('WARNING! %s/%s is not covered under '
-                                   'CODEOWNERS' % (root, file_name))
+                            break
+                    if not match and self.verbose_mode_enabled:
```
If verbose mode is not enabled, we still need to show the viewer which file failed, right? Shouldn't that be stored in a summary message somewhere?

DubeySandeep replied:
Ah, I've done my whole testing in `--verbose` mode, Fixed!

seanlip reviewed (approved):
Generally speaking this LGTM and I think 8-10 secs is OK. Thanks @DubeySandeep!

YashJipkate commented on .gitignore:
```diff
@@ -3,6 +3,7 @@
 *.swp
 *.swo
 gae_runtime/*
+.git/*
```
I don't get this change? Was this intended? If yes, could you please elaborate. This contains the git information and putting this in the gitignore seems a little off.

DubeySandeep replied:
I've added a comment hope that helps, let me know if it's still not clear. Thanks!

YashJipkate replied:
Ah, Ok, that helps. Thanks for the explaination!

YashJipkate commented on scripts/pre_commit_linter.py:
```diff
@@ -2075,93 +2135,36 @@ def _check_codeowner_file(self):
-                                   'CODEOWNERS' % (root, file_name))
+                            break
+                    if not match and self.verbose_mode_enabled:
+                        print '%s is not covered under CODEOWNERS' % file_path
```
I tried deleting a codeowner rule and ran the linter, it didn't catch it and passed the tests as usual. Is this expected behaviour?

DubeySandeep replied:
It depends on which rule you have deleted, assume someone is looking for oppia/storage and you have deleted oppia/storage/exploration, in this case, the test will not fail. Can you please tell which line you have deleted?

YashJipkate replied:
I have deleted [L#170](https://github.com/oppia/oppia/pull/6616/files#diff-5cfb934ca0ec4c1743905a51dc428b84R170). And the checks pass. Let me know if I am missing something here.

DubeySandeep replied:
I tried the same and got this:
![image](https://user-images.githubusercontent.com/16653571/56899221-8cc4b200-6ab0-11e9-942c-54cd3d0809d7.png)

Can you please add the steps you are following?

YashJipkate replied:
Ah never mind, the checks failed as expected. I pulled this PR's latest state and that way it worked!
Thanks!

YashJipkate reviewed (changes requested):
Thanks, @DubeySandeep just some doubts. PTAL!

DubeySandeep committed:
Added review changes and added comments wherever required.

DubeySandeep commented:
@apb7, Can you please review this PR as a code owner?

DubeySandeep commented:
@apb7 & @YashJipkate PTAL!

YashJipkate commented:
@DubeySandeep Sorry for the delay; I have addressed the comments. PTAL!
Thanks

YashJipkate reviewed (approved):
This LGTM!
Thanks @DubeySandeep

DubeySandeep committed:
Updated with upstream/develop.

DubeySandeep committed:
Changed test file name.

DubeySandeep committed:
Fixed filename issues in CODEOWNERS file.

DubeySandeep commented:
@apb7, can you please review this PR?

DubeySandeep commented:
Hi @apb7, PTAL!

DubeySandeep commented:
@apb7, can we merge this PR?

apb7 commented on scripts/pre_commit_linter.py:
```diff
@@ -473,6 +468,56 @@ def _get_expression_from_node_if_one_exists(
+    """
+    command = ['git', 'check-ignore', '-q', path_to_check]
+
+    # The "git check-ignore <path>" command use to returns 0 when the path is
```
Can you please re-phrase this comment?

DubeySandeep replied:
I'll surely do it. Also, it would be great if you can provide some suggestion on what I should add or a pattern in which I can re-phrase this comment! :)

apb7 replied:
Maybe something on these grounds:
`The "git check-ignore <path>" command returns 0 when the path is ignored otherwise it returns 1. subprocess.call then returns this returncode.`

DubeySandeep replied:
Done!

apb7 reviewed (approved):
LGTM just one comment above. Thanks, @DubeySandeep!

vinitamurthi commented on .github/CODEOWNERS:
```diff
@@ -130,85 +226,127 @@
+/core/templates/dev/head/pages/practice_session/ @vinitamurthi
 /core/templates/dev/head/pages/question_editor/ @aks681
 /core/templates/dev/head/pages/questions_list/ @aks681
 
```
Can we add question_player in this with me as the code owner? It was added in #6620

DubeySandeep replied:
Thanks for raising this @vinitamurthi, I think within last  28 days we have added many new files so I'll add all those files/folders before merging this PR! (otherwise, the lint test will start failing on develop.)

Currently, I've added a label `PR: don't merge - HAS MERGE CONFLICTS` to avoid merge.

DubeySandeep replied:
Done!

DubeySandeep committed:
Merge branch 'develop' of https://github.com/oppia/oppia into full-codeowners

DubeySandeep committed:
Added codeowners to new files.

DubeySandeep committed:
Added proper comment as suggested by reviewer.

DubeySandeep committed:
Merge branch 'develop' of https://github.com/oppia/oppia into full-codeowners
